### ISSUE TITLE
No Authentication Key and Token

### ISSUE THREAD

**Issue by jeburkha** [reneklootwijk/node-mideahvac#18]
Hi

Midea-discover finds my appliance, but when I try to get an authentication key and token, I get:
- Authentication Key: An unknown error occurred
- Authentication Token: An unknown error occurred

The credentials work on the SmartHome App on my Android phone, but the App is now called SmartHome, not MSmartHome any more. Would appreciate a hint!

**Comment by svobol13** [comment]
@jeburkha I simply solved this issue buy buying:
- ESP8266 ESP-01S wifi module
- ESP8266 adapter 3.3V to 5V for ESP-01 
- ESP-01 USB ESP8266 serial for programming (there are versions with programming switch and without - I got without and then followed instructions in this video to add jumpers: https://www.youtube.com/watch?v=6uaIWZCRSz8&t=187s)
- Jumper cables to connect adapter with AC

⚠️ Do not connect the ESP8266 to AC with the programmer USB module. You have to use the 3.3V to 5V adapter (I damaged one board by not doing this). The programmer is used only for flashing.

![image](https://github.com/reneklootwijk/node-mideahvac/assets/5737461/8696465d-7764-4279-8b63-9041d8aab588)
![image](https://github.com/reneklootwijk/node-mideahvac/assets/5737461/54959a2d-a191-4b90-a0d2-6bb75f03e8f2)
![image](https://github.com/reneklootwijk/node-mideahvac/assets/5737461/993e3f69-585b-4fb6-810d-677f6738a5b9)
![image](https://github.com/reneklootwijk/node-mideahvac/assets/5737461/fb7fbdb9-7036-403e-b9ae-2b0c5ac6711f)

**Comment by jeburkha** [comment]
Yes, I know, I already implemented this workaround. But it doesn't solve the problem that it is not possible any more to get an authentication key and token from Smart Home.

**Comment by reneklootwijk** [comment]
I had another look at the code and I think I fixed the issue. Would be great if you can test.

**Comment by jeburkha** [comment]
Ok, great! I will do the test, but it will take some time, because i first have to change the interface...

**Comment by reneklootwijk** [comment]
You can just run the discover command to obtain token and key, no need to change anything on your side.

**Comment by jeburkha** [comment]
Does the discover comment work without an interface in my network? Because I changed the interface to the self built one...
If not, I first have to change the interface in my AC!

**Comment by reneklootwijk** [comment]
Yes

**Comment by reneklootwijk** [comment]
BTW in my experience a DIY dongle is far more table than the original dongle. The original dongle periodically loses its WiFi connection and needs to be power cycled to recover again.

**Comment by jeburkha** [comment]
Yes, I use the diy interface at the moment, but it's my request after all, so I'll do the test when I get the chance. But please be patient!

**Comment by svobol13** [comment]
@jeburkha upgraded to 0.3.1 and unfortunately it does not work for me with same error as before.

**Comment by reneklootwijk** [comment]
Maybe the issue you experienced was due to a breaking change in node > 20.x where the dns lookup fails. Support for node version > 20.x has been added in version 0.3.3.

**Comment by reneklootwijk** [comment]
No response received
